### PR TITLE
New feature: hold/tap flavor tap-unless-interrupted

### PR DESCRIPTION
**Pull request by jmding8**
While discussing https://github.com/zmkfirmware/zmk/pull/835, it became clear that `Positional Hold/Tap` would likely be quite useful in combination with a new flavor of hold/tap, `tap-unless-interrupted`

This flavor produces a tap behavior in all situations, except if another key is pressed before `tapping-term-ms` expires.

This is very useful with `Positional Hold/Tap` because:
- decide-as-hold-upon-other key-down is, in some ways, a strongly hold-preferring pattern because it is timing agnostic
- however, `Position Hold/Tap` gives us a mechanism to reject unwanted decide-as-hold-upon-other key-down events on the basis of position, which is independent from timing

### CODE REVIEW

File: app/src/behaviors/behavior_hold_tap.c [hunk 1]
```diff
@@ -611,4 +637,4 @@ static struct behavior_hold_tap_data behavior_hold_tap_data;
 
 DT_INST_FOREACH_STATUS_OKAY(KP_INST)
 
-#endif /* DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT) */
+#endif /* DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT) */
```
Comment (okke-formsma) [inline app/src/behaviors/behavior_hold_tap.c:640]: newline

**Comment by okke-formsma** [comment]
lgtm, test fails though

File: app/src/behaviors/behavior_hold_tap.c [hunk 2]
```diff
@@ -1,4 +1,5 @@
 /*
+
```
Comment (okke-formsma) [inline app/src/behaviors/behavior_hold_tap.c:2]: additional newline should be removed

**Comment by petejohanson** [comment]
Can you please rebase now that #1007 is in? Thanks!

**Comment by jmding8** [comment]
sure thing. merged in main, which includes #1007

**Review by petejohanson — APPROVED** [review]
Excellent. Thanks!

**Comment by petejohanson** [comment]
I will squash and merge this time to clean up the commit messages, but in the future, please a [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) approach to your messages, as that will simplify things when we merge them.

Thanks again!

**Comment by jmding8** [comment]
Closed, because #1018 (a cleaner copy of this) was merged

**Pull request closed without merging**